lib/include/openamp/rpmsg.h

```diff
@@ -146,8 +146,8 @@ int rpmsg_send_offchannel_raw(struct rpmsg_endpoint *ept, uint32_t src,
  */
 static inline unsigned int is_rpmsg_ept_ready(struct rpmsg_endpoint *ept)
 {
-	return (ept->dest_addr != RPMSG_ADDR_ANY) &&
-		(ept->addr != RPMSG_ADDR_ANY);
+	return ept && ept->rdev && ept->dest_addr &&
+		ept->dest_addr != RPMSG_ADDR_ANY;
```

Comment:
Same remark here, this break the legacy

Reply:
If the demo code want to use 1-1023, the code should call rpmsg_init_ept instead. Here is a summary what I understand:
1.Well known service call rpmsg_init_ept with address inside 1-1023
2.Other service call rpmsg_create_ept with address >= 1024
3.It's prefer to call rpmsg_create_ept with RPMSG_ADDR_ANY(~0)
From my point, the quality of example code is very pool because it demo the bad practice:
1.Use the reserved address
2.Use the fixed address

Reply:
Right use RPMSG_ADDR_ANY for the source address in demo is a best usage than fixing the endpoint. But we have to support both. So a good compromize would be to  update only a part of the demos. For instance the echo demo.

Reply:
I will provide a patch change ALL 0 to RPMSG_ADDR_ANY.

Reply:
Here is the PR:
https://github.com/OpenAMP/open-amp/pull/207